tycho-sbom/src/main/java/org/eclipse/tycho/sbom/TychoProjectDependenciesConverter.java

```diff
@@ -117,35 +120,37 @@ private void convertToDependency(DependencyTreeNode node, Set<Dependency> depend
 	private List<String> getBomRepresentation(IInstallableUnit iu) {
 		final MavenSession mavenSession = legacySupport.getSession();
 		final List<MavenProject> reactorProjects = mavenSession.getAllProjects();
-		// mutable!
-		List<String> bomRefs = new ArrayList<>();
-		// (I) IU describes local reactor project
-		try {
-			ProjectDependencyClosure dependencyClosure = dependencyProcessor
-					.computeProjectDependencyClosure(reactorProjects, mavenSession);
-			MavenProject iuProject = dependencyClosure.getProject(iu).orElse(null);
-			if (iuProject != null) {
-				String bomRef = modelConverter.generatePackageUrl(iuProject.getArtifact());
-				if (bomRef == null) {
-					LOG.error("Unable to calculate BOM for: " + iuProject);
+		return bomRepresentations.computeIfAbsent(iu, ignore -> {
```

Comment:
why not move session + reactor project inside the computation?

Reply:
No particular reason. I just overlooked that part.